### PR TITLE
[Debt] Improve consistency of best practice application

### LINKED ISSUE #30: [Debt] Improve consistency and best practices across examples
- Use `?` instead of `.unwrap()` across all examples
- `url` -> `rpc_url`
- Refer to `tx_request` as `tx`, `pending` as `pending_tx`
- Refer to transaction sending as `sending`, not `broadcast`
- Consistently group code blocks with explaining comment - treat each example as if it is to be understood on its own. Each example should have more detailed comments at the code block it focuses on, less on the setup.
- In most cases, refer to `signer` instead of `wallet`
- Use Alice and Bob if referring to a `from` and `to` pair, if singular `from` prefer unnamed

### PR DESCRIPTION
**Pull request by zerosnacks**
Closes #30 

Apologies for the large file diff, it was unavoidable.

Largely cosmetic changes aimed at increasing consistency across examples.
This helps create a basic style guide to roughly follow for outside contributors.

- Adds `clippy` lint rules in workspace root, use the same configuration as Reth. In addition warn on `uninlined_format_args`
- Use consistent way of setting up example environment (anvil, provider) where applicable
- Use `?` instead of `.unwrap()` across all examples
- `url` -> `rpc_url`
- Refer to `tx_request` as `tx`, `pending` as `pending_tx`
- Refer to transaction sending as `sending`, not `broadcast`
- Consistently group code blocks with explaining comment - treat each example as if it is to be understood on its own. Each example should have more detailed comments at the code block it focuses on, less on the setup.
- In most cases, refer to `signer` instead of `wallet`
- Use Alice and Bob if referring to a `from` and `to` pair, if singular `from` prefer unnamed
- End comments with a `.`

Should contain no breaking changes and fundamental changes in behaviour.

### CODE REVIEW

File: examples/transactions/examples/trace_transaction.rs [hunk 1]
```diff
@@ -15,20 +15,23 @@ use eyre::Result;
...
 
-    let url = anvil.endpoint().parse().unwrap();
-    let provider = HttpProvider::<Ethereum>::new_http(url);
-    let hash = fixed_bytes!("97a02abf405d36939e5b232a5d4ef5206980c5a6661845436058f30600c52df7"); // Hash of the tx we want to trace
+    // Create a provider.
+    let rpc_url = anvil.endpoint().parse()?;
+    let provider = HttpProvider::<Ethereum>::new_http(rpc_url);
+
+    // Hash of the tx we want to trace
+    let hash = fixed_bytes!("97a02abf405d36939e5b232a5d4ef5206980c5a6661845436058f30600c52df7");
```
Comment (DaniPopes) [inline examples/transactions/examples/trace_transaction.rs:26]: nit: `b256!`

File: examples/transactions/examples/trace_transaction.rs [hunk 2]
```diff
@@ -15,20 +15,23 @@ use eyre::Result;
...
 
     // Default tracing
     let default_options = GethDebugTracingOptions::default();
-    let res = provider.debug_trace_transaction(hash, default_options).await?;
+    let result = provider.debug_trace_transaction(hash, default_options).await?;
 
-    println!("DEFAULT_TRACE: {:?}", res);
+    println!("DEFAULT_TRACE: {:?}", result);
```
Comment (DaniPopes) [inline examples/transactions/examples/trace_transaction.rs:32]: nit: inline into format string, can run `cargo clippy --fix -- --warn clippy::uninlined_format_args`

**Review by DaniPopes — APPROVED** [review]
nice

File: examples/transactions/examples/trace_transaction.rs (hunk 2 above)
Comment (zerosnacks) [reply examples/transactions/examples/trace_transaction.rs]: Fixed 👍 

Added the lint rules from Reth to apply this consistently going forward

**Pull request merged**